app/src/androidTestGecko/java/org/mozilla/focus/privacy/WebViewDataTest.java

```diff
@@ -0,0 +1,308 @@
+/* -*- Mode: Java; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: nil; -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.privacy;
+
+import android.content.Context;
+import android.preference.PreferenceManager;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+import android.support.test.uiautomator.UiObject;
+import android.support.test.uiautomator.UiObjectNotFoundException;
+import android.support.test.uiautomator.UiSelector;
+
+import android.util.Log;
+import junit.framework.Assert;
+
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.RecordedRequest;
+import org.mozilla.focus.activity.MainActivity;
+import org.mozilla.focus.helpers.TestHelper;
+
+import static android.support.test.espresso.action.ViewActions.click;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mozilla.focus.helpers.TestHelper.waitingTime;
+import static org.mozilla.focus.fragment.FirstrunFragment.FIRSTRUN_PREF;
+
+/**
+ * This test browses to a test site and makes sure that no traces are left on disk.
+ *
+ * The test uses a whitelist so it might fail as soon as you store new files on disk.
+ */
+@RunWith(AndroidJUnit4.class)
+public class WebViewDataTest {
```

Comment:
\[PMD\] INFO: The class 'WebViewDataTest' has a Cyclomatic Complexity of 3 \(Highest = 5\)\.